<div class="text">

<h2>Test datasets</h2>
<p>Every test dataset is a single FASTA-formatted file.
We tried to include broad variety of commonly used DNA, RNA and protein datasets.
Suggestions for additional interesting datasets are welcome!</p>

<p>All data is available for download at the <span class="arrow">&#x2193;</span> links (in the <a href="https://github.com/KirillKryukov/naf">naf</a> format).</p>

<h3>Genomes</h3>
<p>Genomes are DNA datasets.
Typically they have long sequences with relatively few repetitions, and compress poorly.
All genomes we used are from <a href="https://www.ncbi.nlm.nih.gov/assembly/">NCBI Assembly</a>.
These genomes were selected more or less randomly with the purpose of covering variety of sizes.</p>
<table class="d" style="margin: 0 auto;">
<thead><tr><td>Category</td><td>Organism</td><td>Accession</td><td>Size</td><td class="dl"><span class="arrow">&#x2193;</span></td><td>Source</td></tr></thead>
<tbody>
<tr><td>Virus</td><td><i>Gordonia phage GAL1</i></td><td>GCF_001884535.1</td><td>50.7 kB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Gordonia-phage-GAL1-GCF_001884535.1-2016-11-15.naf"><table><tr><td class="arrow">&#x2193;</td><td>(12.4&nbsp;kB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/884/535/GCF_001884535.1_ViralProj353700/GCF_001884535.1_ViralProj353700_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Bacteria</td><td><i>WS1 bacterium JGI 0000059-K21</i></td><td>GCA_000398605.1</td><td>522 kB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-WS1-bacterium-JGI-0000059-K21-GCA_000398605.1-2013-05-16.naf"><table><tr><td class="arrow">&#x2193;</td><td>(125&nbsp;kB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/398/605/GCA_000398605.1_ASM39860v1/GCA_000398605.1_ASM39860v1_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Protist</td><td><i>Astrammina rara</i></td><td>GCA_000211355.2</td><td>1.71 MB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Astrammina-rara-GCA_000211355.2-2011-04-27.naf"><table><tr><td class="arrow">&#x2193;</td><td>(384&nbsp;kB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/211/355/GCA_000211355.2_ASM21135v2/GCA_000211355.2_ASM21135v2_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Fungi</td><td><i>Nosema ceranae</i></td><td>GCA_000988165.1</td><td>5.81 MB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Nosema-ceranae-GCA_000988165.1-2015-05-05.naf"><table><tr><td class="arrow">&#x2193;</td><td>(1.37&nbsp;MB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/988/165/GCA_000988165.1_ASM98816v1/GCA_000988165.1_ASM98816v1_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Protist</td><td><i>Cryptosporidium parvum Iowa II</i></td><td>GCA_000165345.1</td><td>9.22 MB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Cryptosporidium-parvum-Iowa-II-GCA_000165345.1-2007-02-26.naf"><table><tr><td class="arrow">&#x2193;</td><td>(2.29&nbsp;MB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/165/345/GCA_000165345.1_ASM16534v1/GCA_000165345.1_ASM16534v1_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Protist</td><td><i>Spironucleus salmonicida</i></td><td>GCA_000497125.1</td><td>13.1 MB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Spironucleus-salmonicida-GCA_000497125.1-2013-11-19.naf"><table><tr><td class="arrow">&#x2193;</td><td>(3.15&nbsp;MB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/497/125/GCA_000497125.1_SSK3.0/GCA_000497125.1_SSK3.0_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Protist</td><td><i>Tieghemostelium lacteum</i></td><td>GCA_001606155.1</td><td>23.7 MB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Tieghemostelium-lacteum-GCA_001606155.1-2016-04-04.naf"><table><tr><td class="arrow">&#x2193;</td><td>(5.71&nbsp;MB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/606/155/GCA_001606155.1_ASM160615v1/GCA_001606155.1_ASM160615v1_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Fungus</td><td><i>Fusarium graminearum PH-1</i></td><td>GCF_000240135.3</td><td>36.9 MB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Fusarium-graminearum-PH-1-GCF_000240135.3-2008-11-21.naf"><table><tr><td class="arrow">&#x2193;</td><td>(9.32&nbsp;MB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/240/135/GCF_000240135.3_ASM24013v3/GCF_000240135.3_ASM24013v3_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Protist</td><td><i>Salpingoeca rosetta</i></td><td>GCA_000188695.1</td><td>56.2 MB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Salpingoeca-rosetta-GCA_000188695.1-2011-02-10.naf"><table><tr><td class="arrow">&#x2193;</td><td>(12.3&nbsp;MB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/188/695/GCA_000188695.1_Proterospongia_sp_ATCC50818/GCA_000188695.1_Proterospongia_sp_ATCC50818_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Algae</td><td><i>Chondrus crispus</i></td><td>GCA_000350225.2</td><td>106 MB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Chondrus-crispus-GCA_000350225.2-2013-05-22.naf"><table><tr><td class="arrow">&#x2193;</td><td>(16.0&nbsp;MB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/350/225/GCA_000350225.2_ASM35022v2/GCA_000350225.2_ASM35022v2_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Algae</td><td><i>Kappaphycus alvarezii</i></td><td>GCA_002205965.2</td><td>341 MB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Kappaphycus-alvarezii-GCA_002205965.2-2018-03-09.naf"><table><tr><td class="arrow">&#x2193;</td><td>(66.1&nbsp;MB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002/205/965/GCA_002205965.2_ASM220596v2/GCA_002205965.2_ASM220596v2_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Animal</td><td><i>Strongylocentrotus purpuratus</i></td><td>GCF_000002235.4</td><td>1.01 GB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Strongylocentrotus-purpuratus-GCF_000002235.4-2015-03-10.naf"><table><tr><td class="arrow">&#x2193;</td><td>(193&nbsp;MB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/235/GCF_000002235.4_Spur_4.2/GCF_000002235.4_Spur_4.2_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Animal</td><td><i>Homo sapiens</i></td><td>GCA_000001405.28</td><td>3.31 GB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Homo-sapiens-GCA_000001405.28-2019-02-28.naf"><table><tr><td class="arrow">&#x2193;</td><td>(691&nbsp;MB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.28_GRCh38.p13/GCA_000001405.28_GRCh38.p13_genomic.fna.gz">NCBI FTP</a></td></tr>
<tr><td>Plant</td><td><i>Picea abies</i></td><td>GCA_900067695.1</td><td>13.4 GB</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Genome-Picea-abies-GCA_900067695.1-2016-11-09.naf"><table><tr><td class="arrow">&#x2193;</td><td>(2.31&nbsp;GB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/067/695/GCA_900067695.1_Pabies01/GCA_900067695.1_Pabies01_genomic.fna.gz">NCBI FTP</a></td></tr>
</tbody></table>

<h3>Other DNA datasets</h3>
<p>These datasets are highly repetitive,
giving the compressors plenty of redundancy to work with.</p>
<table class="d" style="margin: 0 auto;">
<thead><tr><td>Dataset</td><td>Size</td><td>Number of<br />sequences</td><td>Date</td><td class="dl"><span class="arrow">&#x2193;</span></td><td>Source</td></tr></thead>
<tbody>

<tr><td>Mitochondrion</td><td>245&nbsp;MB</td><td>9,402</td><td>2019-03-15</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Mitochondrion-2019-03-15.naf"><table><tr><td class="arrow">&#x2193;</td><td>(36.1&nbsp;MB)</td></tr></table></a></td>
    <td>Collection of mostly complete mitochondrial genomes from NCBI
        (<a href="ftp://ftp.ncbi.nlm.nih.gov/refseq/release/mitochondrion/mitochondrion.1.1.genomic.fna.gz">1</a>,
         <a href="ftp://ftp.ncbi.nlm.nih.gov/refseq/release/mitochondrion/mitochondrion.2.1.genomic.fna.gz">2</a>).</td></tr>

<tr><td>NCBI Virus Complete Nucleotide Human</td><td>482&nbsp;MB</td><td>36,745</td><td>2020-05-11</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-NCBI-Virus-Complete-Nucleotide-Human-2020-05-11.naf"><table><tr><td class="arrow">&#x2193;</td><td>(9.27&nbsp;MB)</td></tr></table></a></td>
    <td>From <a href="https://www.ncbi.nlm.nih.gov/labs/virus/vssi/">NCBI Virus</a></td></tr>

<tr><td>Influenza</td><td>1.22&nbsp;GB</td><td>700,001</td><td>2019-04-27</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Influenza-2019-04-27.naf"><table><tr><td class="arrow">&#x2193;</td><td>(13.8&nbsp;MB)</td></tr></table></a></td>
    <td>Entire set of sequences from the <a href="https://www.ncbi.nlm.nih.gov/genomes/FLU/Database/nph-select.cgi?go=database">Influenza Virus Database</a>
        (<a href="ftp://ftp.ncbi.nih.gov/genomes/INFLUENZA/influenza.fna.gz">source</a>).</td></tr>

<tr><td>Helicobacter</td><td>2.76&nbsp;GB</td><td>108,292</td><td>2019-04-24</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-Helicobacter-2019-04-24.naf"><table><tr><td class="arrow">&#x2193;</td><td>(130&nbsp;MB)</td></tr></table></a></td>
    <td>1,622 genomes of <i>Helicobacter</i> (including <i>H. pylori</i>), all available at GenBank and RefSeq as of 2019-04-24,
        obtained from <a href="https://www.ncbi.nlm.nih.gov/assembly">NCBI Assembly</a>.</td></tr>

</tbody></table>

<h3>RNA datasets</h3>
<p>These are highly repetitive single gene RNA databases.</p>
<table class="d" style="margin: 0 auto;">
<thead><tr><td>Dataset</td><td>Size</td><td>Number of<br />sequences</td><td>Date</td><td class="dl"><span class="arrow">&#x2193;</span></td><td>Source</td></tr></thead>
<tbody>

<tr><td>SILVA 132 LSURef</td><td>610&nbsp;MB</td><td>198,843</td><td>2017-12-11</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/RNA-SILVA-132-LSURef-2017-12-11.naf"><table><tr><td class="arrow">&#x2193;</td><td>(12.6&nbsp;MB)</td></tr></table></a></td>
    <td><a href="https://ftp.arb-silva.de/release_132/Exports/SILVA_132_LSURef_tax_silva.fasta.gz">source link</a></td></tr>

<tr><td>SILVA 132 SSURef Nr99</td><td>1.11&nbsp;GB</td><td>695,171</td><td>2017-12-11</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/RNA-SILVA-132-SSURef-Nr99-2017-12-11.naf"><table><tr><td class="arrow">&#x2193;</td><td>(41.7&nbsp;MB)</td></tr></table></a></td>
    <td><a href="https://ftp.arb-silva.de/release_132/Exports/SILVA_132_SSURef_Nr99_tax_silva.fasta.gz">source link</a></td></tr>

<tr><td>SILVA 132 SSURef</td><td>3.28&nbsp;GB</td><td>2,090,668</td><td>2017-12-11</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/RNA-SILVA-132-SSURef-2017-12-11.naf"><table><tr><td class="arrow">&#x2193;</td><td>(78.1&nbsp;MB)</td></tr></table></a></td>
    <td><a href="https://ftp.arb-silva.de/release_132/Exports/SILVA_132_SSURef_tax_silva.fasta.gz">source link</a></td></tr>

</tbody></table>

<h3>Multiple Sequence Alignments</h3>
<p>Aligned DNA sequences stored in FASTA format.</p>
<table class="d" style="margin: 0 auto;">
<thead><tr><td>Dataset</td><td>Size</td><td>Number of<br />sequences</td><td>Date</td><td class="dl"><span class="arrow">&#x2193;</span></td><td>Source</td></tr></thead>
<tbody>

<tr><td>UCSC hg38 7way knownCanonical-exonNuc</td><td>340&nbsp;MB</td><td>1,470,154</td><td>2014-06-06</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-UCSC-hg38-7way-knownCanonical-exonNuc-2014-06-06.naf"><table><tr><td class="arrow">&#x2193;</td><td>(44.5&nbsp;MB)</td></tr></table></a></td>
    <td>Alignments of 6 vertebrate genomes with human for CDS regions. Downloaded from <a href="https://hgdownload.soe.ucsc.edu/downloads.html">UCSC</a>,
        <a href="https://hgdownload.soe.ucsc.edu/goldenPath/hg38/multiz7way/alignments/knownCanonical.exonNuc.fa.gz">source</a>
        (removed empty lines to normalize the FASTA format).</td></tr>

<tr><td>UCSC hg38 20way knownCanonical-exonNuc</td><td>969&nbsp;MB</td><td>4,211,940</td><td>2015-06-30</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/DNA-UCSC-hg38-20way-knownCanonical-exonNuc-2015-06-30.naf"><table><tr><td class="arrow">&#x2193;</td><td>(75.9&nbsp;MB)</td></tr></table></a></td>
    <td>Alignments of 19 mammalian genomes with human for CDS regions. Downloaded from <a href="https://hgdownload.soe.ucsc.edu/downloads.html">UCSC</a>,
        <a href="https://hgdownload.soe.ucsc.edu/goldenPath/hg38/multiz20way/alignments/knownCanonical.exonNuc.fa.gz">source</a>
        (removed empty lines to normalize the FASTA format).</td></tr>
</tbody></table>

<h3>Protein databases</h3>
<p>Protein data compresses more poorly than DNA/RNA (and sometimes even called <a href="https://doi.org/10.1109/DCC.1999.755675">incompressible</a>).
However they are also important datasets.
Unfortunately most of sequence compressors don't support protein data.</p>
<table class="d" style="margin: 0 auto;">
<thead><tr><td>Dataset</td><td>Size</td><td>Number of<br />sequences</td><td>Date</td><td class="dl"><span class="arrow">&#x2193;</span></td><td>Source</td></tr></thead>
<tbody>

<tr><td>PDB</td><td>67.6&nbsp;MB</td><td>109,914</td><td>2019-04-09</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/AA-PDB-2019-04-09.naf"><table><tr><td class="arrow">&#x2193;</td><td>(10.1&nbsp;MB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ncbi.nih.gov/blast/db/FASTA/pdbaa.gz">source link</a></td></tr>

<tr><td>Homo sapiens GRCh38 peptides all</td><td>73.2&nbsp;MB</td><td>105,961</td><td>2019-03-12</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/AA-Homo-sapiens-GRCh38-peptides-all-2019-03-12.naf"><table><tr><td class="arrow">&#x2193;</td><td>(8.47&nbsp;MB)</td></tr></table></a></td>
    <td><a href="ftp://ftp.ensembl.org/pub/release-96/fasta/homo_sapiens/pep/Homo_sapiens.GRCh38.pep.all.fa.gz">source link</a></td></tr>

<tr><td>NCBI Virus RefSeq Protein</td><td>122&nbsp;MB</td><td>373,332</td><td>2020-05-10</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/AA-NCBI-Virus-RefSeq-Protein-2020-05-10.naf"><table><tr><td class="arrow">&#x2193;</td><td>(29.9&nbsp;MB)</td></tr></table></a></td>
    <td>From <a href="https://www.ncbi.nlm.nih.gov/labs/virus/vssi/">NCBI Virus</a></td></tr>

<tr><td>UniProtKB Reviewed (Swiss-Prot)</td><td>277&nbsp;MB</td><td>560,118</td><td>2019-04-02</td>
    <td class="dl"><a href="http://kirill.med.u-tokai.ac.jp/sequence-compression-benchmark-test-data/AA-UniProtKB-Reviewed-Swiss-Prot-2019-04-02.naf"><table><tr><td class="arrow">&#x2193;</td><td>(57.4&nbsp;MB)</td></tr></table></a></td>
    <td>From <a href="https://www.uniprot.org/">UniProt</a>,
        <a href="https://www.uniprot.org/news/2019/05/08/release">release&nbsp;2019_04</a>,
        <a href="ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz">source&nbsp;link</a></td></tr>

</tbody></table>

</div>
